lib/sensors/led_driver.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "i2c_sensors.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+
+#include <core/logger.hpp>
+#include <io/i2c.hpp>
+
+int kDefaultLedDriverAddress = 0x30;
+
+namespace hyped::sensors {
+class LedDriver {
+ public:
+  static std::optional<LedDriver> create(core::ILogger &logger,
+                                         std::shared_ptr<io::II2c> i2c,
+                                         const std::uint8_t device_address
+                                         = kDefaultLedDriverAddress);
+  ~LedDriver();
+
+  std::optional<core::Result> initialise();
+  std::optional<core::Result> set_colour(std::uint8_t channel,
+                                         std::uint8_t red,
+                                         std::uint8_t green,
+                                         std::uint8_t blue);
+  std::optional<core::Result> set_intensity(std::uint8_t channel, std::uint8_t intensity);
+  std::optional<core::Result> reset();
```

Comment:
can you write a docstring for all of these methods, especially reset as it is a little unclear what it does